sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -102,6 +102,83 @@ abstract class Catalog {
   def listColumns(dbName: String, tableName: String): Dataset[Column]
 
   /**
+   * Find the database with the specified name. This returns [[None]] when no [[Database]] can be
+   * found.
+   *
+   * @since 2.1.0
+   */
+  def findDatabase(dbName: String): Option[Database]
```

Comment:
Sorry, I'm a bit late to this party.  In terms of Scala vs. Java API, "find*" doesn't really strike me as the best naming, especially not after these "find*" methods no longer produce an Option.  In Scala, I expect a `find` to take a predicate and to produce an Option that may contain the first element to satisfy the predicate.  Something that throws an exception when the one thing asked for doesn't exist feels more like a `get` than a `find` to me.

Can we reconsider before we get to the point where we find that this API is set in stone?